From 403eee165c7763c56ba8b51ad91b4b6cf0811b40 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Tue, 21 Jun 2011 01:11:28 +0200
Subject: [PATCH 01/57] h264: Fix assert that failed to compile with -DDEBUG.

The assert referenced a variable that no longer exists since 4:4:4 support.
(cherry picked from commit 6371ce4b0f0535a7f576b8f109d35345e3d11d7d)

Signed-off-by: Reinhard Tartler <siretart@tauware.de>
---
 libavcodec/h264.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index dc4acb8..13a6380 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -1682,7 +1682,7 @@ static av_always_inline void hl_decode_mb_predict_luma(H264Context *h, int mb_ty
                         uint64_t tr_high;
                         if(dir == DIAG_DOWN_LEFT_PRED || dir == VERT_LEFT_PRED){
                             const int topright_avail= (h->topright_samples_available<<i)&0x8000;
-                            assert(mb_y || linesize <= block_offset[i]);
+                            assert(s->mb_y || linesize <= block_offset[i]);
                             if(!topright_avail){
                                 if (pixel_shift) {
                                     tr_high= ((uint16_t*)ptr)[3 - linesize/2]*0x0001000100010001ULL;
-- 
1.7.4.1

